Add ci deploy (#1076)
alphagov/datagovuk_publish: README.md

This repository contains the beta-stage publishing component of data.gov.uk.

# Deployment

> Continuous Integration has been setup using github actions. 

Reviewer: ```suggestion
Continuous Integration has been setup using GitHub Actions.
```

> Tests are run on pull requests, deployments to Staging are via merges of branches into the `main` branch.

Reviewer: ```suggestion
Tests are run on pull requests. Deployments to Staging happen automatically when merging branches into `main`.
```

> In order to carry out a release to production a developer in the govuk team will need to create a release tag that leads with a `v` and approve of the deployment in github actions.

Reviewer: I didn't understand "approve of the deployment in github actions" - I don't think GitHub Actions are involved in this step. I've rewritten what I think you meant below:

```suggestion
To deploy to Production, you'll need to [publish a Release](https://github.com/alphagov/datagovuk_publish/releases/new), specifying a "Tag version" beginning with `v` (e.g. `v1.0.0`).
```

Author: There is an additional step to approve a deployment, this is to protect secrets from leaking out as the `production` secrets cannot be protected with tagged releases.

Further information about the deploying to PaaS are in the developer documents here: 

> https://docs.publishing.service.gov.uk/manual/data-gov-uk-deployment.html#paas-staging-and-production-environments

Reviewer: Out of scope, but could the contents of that document be brought into the `docs/` folder of this repo? That way we can keep all the documentation here in one place (helping it stay up to date), and it will also be auto-imported into the Developer Docs.

Author: I think the reason it's there is so that there's only one place where that information needs to be updated.
